packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js

```diff
@@ -901,9 +903,100 @@ const Conversation = WebexPlugin.extend({
     this.webex.trigger('user-activity');
 
     return this.request(params)
-      .then((res) => res.body);
+      .then((res) => res.body)
+      .catch((error) => {
+        // handle when key need to rotate
+        if (error.body && error.body.errorCode === KEY_ROTATION_REQUIRED) {
+          this.webex.request({
+            method: 'GET',
+            api: 'conversation',
+            resource: `conversations/${params.body.target.id}`
+          }).then((res) => {
+            res.body.defaultActivityEncryptionKeyUrl = null;
+            if (activity.object.files.items.length > 0) {
+              this.resendShareActivity(res, activity);
+            }
+            else {
+              this.resendPostActivity(res, activity);
+            }
+          });
+        }
+        else if (
+          error.body &&
+    (error.body.errorCode === KEY_ALREADY_ROTATED || error.body.errorCode === ENCRYPTION_KEY_URL_MISMATCH)
+        ) {
+          // handle when key need to update
+          this.webex.request({
+            method: 'GET',
+            api: 'conversation',
+            resource: `conversations/${params.body.target.id}`
+          }).then((res) => {
```

Comment:
This GET request makes sense since we need to get the new key

Reply:
Could we use the `this.get` function instead of creating an ad hoc request here?

Reply:
get function take conversation as parameter but in failed request not having complete convo object.